pkgs/applications/window-managers/hyprland/default.nix

```diff
@@ -0,0 +1,77 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, libdrm
+, libinput
+, libxcb
+, libxkbcommon
+, mesa
+, pango
+, wayland
+, wayland-protocols
+, wayland-scanner
+, wlroots
+, xcbutilwm
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hyprland";
+  version = "0.6.0beta";
+
+  # When updating Hyprland, the overridden wlroots commit must be bumped to match the commit upstream uses.
+  src = fetchFromGitHub {
+    owner = "hyprwm";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-SHElAr8ZnLirEX8qd4vRyNaorUMi0lAEn/v5+GIlA5c=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    wayland-scanner
+  ];
+
+  buildInputs = [
+    libdrm
+    libinput
+    libxcb
+    libxkbcommon
+    mesa
+    pango
+    wayland
+    wayland-protocols
+    wlroots
+    xcbutilwm
+  ];
+
+  # build with system wlroots
+  postPatch = ''
+    sed -Ei 's/"\.\.\/wlroots\/include\/([a-zA-Z0-9./_-]+)"/<\1>/g' src/includes.hpp
+
+    make protocols
```

Comment:
This should be done in some configure or build phase. I am not sure which is more correct but doing it in patchPhase is not the correct place.